dlib/dnn/input.h

```diff
@@ -591,30 +591,13 @@ namespace dlib
 // ----------------------------------------------------------------------------------------
 
     template <typename PYRAMID_TYPE>
-    class input_rgb_image_pyramid
+    class input_image_pyramid
```

Comment:
This should probably go in a `detail` namespace to discourage users from touching it.